### PR TITLE
Tensor.uniform with dtype=int bug fix

### PR DESCRIPTION
**Pull request by jaw566**
Identified the following bug: invoking Tensor.uniform with dtype=int32 returns a tensor with every entry equal to the low argument instead of a set of uniformly distributed integers

### CODE REVIEW

**Comment by tinyb0t** [comment]
Changes made in `tinygrad/`:
```
------------------------------------------------------------
files                             insertions       deletions
------------------------------------------------------------
tinygrad/lazy.py                           1               1
tinygrad/tensor.py                         2               1
------------------------------------------------------------
total                                      3               2
------------------------------------------------------------
lines added in the tinygrad folder: 1
```

File: tinygrad/tensor.py [hunk 1]
```diff
@@ -182,7 +182,8 @@ def randn(*shape, dtype:Optional[DType]=None, **kwargs) -> Tensor:
   def normal(*shape, mean=0.0, std=1.0, **kwargs) -> Tensor: return (std * Tensor.randn(*shape, **kwargs)) + mean
 
   @staticmethod
-  def uniform(*shape, low=-1.0, high=1.0, **kwargs) -> Tensor: return ((high-low) * Tensor.rand(*shape, **kwargs)) + low
+  def uniform(*shape, low=-1.0, high=1.0, dtype:Optional[DType]=None, **kwargs) -> Tensor:
+    return ((high-low) * Tensor.rand(*shape, **kwargs)).cast(Tensor.default_type if dtype is None else dtype) + low
```
Comment (dc-dc-dc) [inline tinygrad/tensor.py:186]: is dtype not getting passed via kwargs?

Comment (jaw566) [reply tinygrad/tensor.py:186]: Ah, yes, looking into this now

Comment (ZenginU) [reply tinygrad/tensor.py:186]: I believe your approach is kinda true, if you give dtype with kwargs Tensor.rand is converted to 0 or 1 implicitly.

File: tinygrad/tensor.py [hunk 2]
```diff
@@ -182,8 +182,9 @@ def randn(*shape, dtype:Optional[DType]=None, **kwargs) -> Tensor:
   def normal(*shape, mean=0.0, std=1.0, **kwargs) -> Tensor: return (std * Tensor.randn(*shape, **kwargs)) + mean
 
   @staticmethod
-  def uniform(*shape, low=-1.0, high=1.0, dtype:Optional[DType]=None, **kwargs) -> Tensor:
-    return ((high-low) * Tensor.rand(*shape, **kwargs)).cast(Tensor.default_type if dtype is None else dtype) + low
+  def uniform(*shape, low=-1.0, high=1.0, **kwargs) -> Tensor:
+    dtype = kwargs.pop("dtype") if "dtype" in kwargs else Tensor.default_type
```
Comment (ZenginU) [inline tinygrad/tensor.py:186]:
```
pop(key[, default])
If key is in the dictionary, remove it and return its value, else return default. If default is not
```
I believe
```python
dtype = kwargs.pop("dtype", Tensor.default_type)
```
is more readable

File: tinygrad/tensor.py (hunk 1 above)
Comment (jaw566) [reply tinygrad/tensor.py]: I think my latest commit (pruning dtype from kwargs) addresses this issue?

File: tinygrad/tensor.py (hunk 2 above)
Comment (jaw566) [reply tinygrad/tensor.py:186]: Totally agree. Fixing now

**Pull request merged**